### ISSUE TITLE
 refer an alias in last_value() raise an internal error, then database was unavailable

### ISSUE THREAD

**Issue by l1t1** [duckdb/duckdb#8671]
### What happens?

refer an alias in last_value() raise an internal error, then database was unavailable

### To Reproduce

```
v0.8.2-dev3007 dd7f0c0870
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
```
```sql
create table tmp( t_date varchar(1000), data1 int );
insert into tmp values('2023-08-01',40);
insert into tmp values('2023-08-02',40);
insert into tmp values('2023-08-03',40);
insert into tmp values('2023-08-04',10);
insert into tmp values('2023-08-05',10);
insert into tmp values('2023-08-06',40);
insert into tmp values('2023-08-07',40);
insert into tmp values('2023-08-08',30);
insert into tmp values('2023-08-09',30);
insert into tmp values('2023-08-10',30);
```
```sql
select *,row_number()over(order by t_date) a, last_value(a)over(partition by data1)from tmp  order by 1;
```
```text
Error: INTERNAL Error: Failed to bind column reference "a" [5.0] (bindings: [0.0 0.1])
```
```sql 
select *,row_number()over(order by t_date) a, last_value(1)over(partition by data1)from tmp  order by 1;
```
```text
Error: FATAL Error: Failed: database has been invalidated because of a previous fatal error. The database must be restarted prior to being used again.
Original error: "INTERNAL Error: Failed to bind column reference "a" [5.0] (bindings: [0.0 0.1])"
```

### OS:

Linux aarch64

### DuckDB Version:

0.8.2 dev

### DuckDB Client:

CLI

### Full Name:

lutao

### Affiliation:

home

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by robert-manchester** [comment]
hit this too on v0.8.1 and v0.8.2-dev3107 4c2de0a942
```sql
with
runtimes as (
  unpivot (
    VALUES
    (1692924233,1692936589,1692942573,'run.0','noexclude'),
    (1693016117,1693021831,1693023525,'run.3','exclude')
  ) as t('start','pony','end','dir','tag')
  on
    * exclude (dir, tag)
  into
    name instant value ts
)
select
  *,
  lag(ts) over (partition by dir order by ts) as lag,
  ts - lag as diff,
  sum(diff) over (partition by dir order by ts) as total
from
  runtimes
order by
  dir, ts
;
```
```
Error: near line 1: INTERNAL Error: Failed to bind column reference "lag" [37.0] (bindings: [27.0 27.1 27.2 27.3])
```

**Comment by Mytherin** [comment]
Fixed in v0.9.2

```sql
D create table tmp( t_date varchar(1000), data1 int );
D insert into tmp values('2023-08-01',40);
D insert into tmp values('2023-08-02',40);
D insert into tmp values('2023-08-03',40);
D insert into tmp values('2023-08-04',10);
D insert into tmp values('2023-08-05',10);
D insert into tmp values('2023-08-06',40);
D insert into tmp values('2023-08-07',40);
D insert into tmp values('2023-08-08',30);
D insert into tmp values('2023-08-09',30);
D insert into tmp values('2023-08-10',30);
D 
D select *,row_number()over(order by t_date) a, last_value(a)over(partition by data1)from tmp  order by 1;
Error: Binder Error: window function calls cannot be nested
D 
D select *,row_number()over(order by t_date) a, last_value(1)over(partition by data1)from tmp  order by 1;
┌────────────┬───────┬───────┬─────────────────────────────────────────┐
│   t_date   │ data1 │   a   │ last_value(1) OVER (PARTITION BY data1) │
│  varchar   │ int32 │ int64 │                  int32                  │
├────────────┼───────┼───────┼─────────────────────────────────────────┤
│ 2023-08-01 │    40 │     1 │                                       1 │
│ 2023-08-02 │    40 │     2 │                                       1 │
│ 2023-08-03 │    40 │     3 │                                       1 │
│ 2023-08-04 │    10 │     4 │                                       1 │
│ 2023-08-05 │    10 │     5 │                                       1 │
│ 2023-08-06 │    40 │     6 │                                       1 │
│ 2023-08-07 │    40 │     7 │                                       1 │
│ 2023-08-08 │    30 │     8 │                                       1 │
│ 2023-08-09 │    30 │     9 │                                       1 │
│ 2023-08-10 │    30 │    10 │                                       1 │
├────────────┴───────┴───────┴─────────────────────────────────────────┤
│ 10 rows                                                    4 columns │
└──────────────────────────────────────────────────────────────────────┘

```
